src/main/kotlin/app/revanced/patches/shared/gms/BaseGmsCoreSupportPatch.kt

```diff
@@ -114,6 +119,20 @@ abstract class BaseGmsCoreSupportPatch(
         GmsCoreSupportFingerprint.resultOrThrow().mutableClass.methods
             .single { it.name == GET_GMS_CORE_VENDOR_GROUP_ID_METHOD_NAME }
             .replaceInstruction(0, "const-string v0, \"$gmsCoreVendorGroupId\"")
+
+        RequestChecksumFingerprint.resultOrThrow().mutableMethod.apply {
+            val targetIndex = indexOfFirstInstruction {
+                opcode == Opcode.INVOKE_VIRTUAL
+                    && getReference<MethodReference>()?.name == "getPackageName"
+            } + 1
+
+            val targetRegister = getInstruction<OneRegisterInstruction>(targetIndex).registerA
+
+            replaceInstruction(
+                targetIndex,
+                "const-string v$targetRegister, \"com.google.android.youtube\"",
```

Comment:
This patch is used not only on YouTube but also on YouTube Music
Therefore package names should not be hardcoded